python/packages/autogen-agentchat/src/autogen_agentchat/agents/_user_proxy_agent.py

```diff
@@ -14,6 +15,13 @@
 InputFuncType = Union[SyncInputFunc, AsyncInputFunc]
 
 
+# TODO: ainput doesn't seem to play nicely with jupyter.
+async def cancellable_input(prompt: str, cancellation_token: Optional[CancellationToken]) -> Awaitable[str]:
```

Comment:
Rather than special casing input, we should ensure that any sync input function can be canceled if the token is cancelled  (line 80-82) of the original file